src/Evolution/ComputeTags.hpp

```diff
@@ -11,81 +15,50 @@
   using return_type = typename base::type;
   using argument_tags =
-      tmpl::list<AnalyticSolutionTag,
+      tmpl::list<::Tags::AnalyticSolutionOrData,
                  domain::Tags::Coordinates<Dim, Frame::Inertial>, ::Tags::Time>;
+  template <typename AnalyticSolution>
   static void function(
       const gsl::not_null<return_type*> analytic_solution,
-      const typename AnalyticSolutionTag::type& analytic_solution_computer,
+      const AnalyticSolution& analytic_solution_computer,
       const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords,
       const double time) {
-    *analytic_solution =
-        variables_from_tagged_tuple(analytic_solution_computer.variables(
-            inertial_coords, time, AnalyticFieldsTagList{}));
+    if constexpr (std::is_convertible_v<AnalyticSolution*,
+                                        MarkAsAnalyticSolution*>) {
+      *analytic_solution =
+          variables_from_tagged_tuple(analytic_solution_computer.variables(
+              inertial_coords, time, AnalyticFieldsTagList{}));
+    } else {
+      (void)analytic_solution_computer;
+      (void)inertial_coords;
+      (void)time;
+      *analytic_solution = std::nullopt;
+    }
   }
 };
+}  // namespace evolution::Tags
 
-/// @{
-/*!
- * \brief For each `Tag` in `TagsList`, compute its difference from the
- * analytic solution.
- */
-template <size_t VolumeDim, typename AnalyticSolutionTag, typename TagsList>
-struct ErrorsCompute
-    : db::add_tag_prefix<::Tags::Error, ::Tags::Variables<TagsList>>,
-      db::ComputeTag {
-  using base = db::add_tag_prefix<::Tags::Error, ::Tags::Variables<TagsList>>;
-  using return_type = tmpl::type_from<base>;
-
-  using argument_tags = tmpl::append<
-      tmpl::list<AnalyticSolutionTag,
-                 domain::Tags::Coordinates<VolumeDim, Frame::Inertial>,
-                 ::Tags::Time>,
-      TagsList>;
-
-  template <typename AnalyticSolution, typename... ErrorTags,
-            typename... FieldTypes>
-  static constexpr void function(
-      const gsl::not_null<Variables<tmpl::list<ErrorTags...>>*> errors,
-      const AnalyticSolution& analytic_solution_computer,
-      const tnsr::I<DataVector, VolumeDim, Frame::Inertial>& inertial_coords,
-      const double time, const FieldTypes&... fields) {
-    *errors = return_type{get<0>(inertial_coords).size()};
-    const auto helper = [](const auto error, const auto& field) {
-      for (size_t i = 0; i < field.size(); ++i) {
-        (*error)[i] = field[i];
-      }
-    };
-    EXPAND_PACK_LEFT_TO_RIGHT(
-        helper(make_not_null(&get<ErrorTags>(*errors)), fields));
-
-    const auto analytic =
-        variables_from_tagged_tuple(analytic_solution_computer.variables(
-            inertial_coords, time, TagsList{}));
-    *errors -= analytic;
-  }
-};
-/// @}
+/// \cond
+namespace Tags {
+template <size_t Dim, typename FieldTagsList>
+struct is_analytic_solutions<
```

Comment:
What do you need this for? Please add a comment here, briefly mentioning why you need to specialize the template here.